src/main/java/io/r2dbc/postgresql/client/BalancedResolverGroup.java

```diff
@@ -0,0 +1,21 @@
+package io.r2dbc.postgresql.client;
+
+import io.netty.resolver.AddressResolver;
+import io.netty.resolver.AddressResolverGroup;
+import io.netty.resolver.DefaultNameResolver;
+import io.netty.resolver.RoundRobinInetAddressResolver;
+import io.netty.util.concurrent.EventExecutor;
+
+import java.net.InetSocketAddress;
+
+/**
+ * When the {@link InetSocketAddress} resolves to multiple IP addresses, pick one randomly.
+ */
+public class BalancedResolverGroup extends AddressResolverGroup<InetSocketAddress> {
```

Comment:
No need to have this class `public`. Please make it package-private and `final`. Since there's no state associated, it would make sense to keep a static instance (singleton) instead of creating a new instance on each connect.

Reply:
done